### PR TITLE
Add support for `Local` provider for e2e tests

### LINKED ISSUE #667: [Feature] Support `Local` provider for e2e tests
**Feature (What you would like to be added):**
Support `Local` backup bucket provider for e2e tests.

**Motivation (Why is this needed?):**
Currently, e2e tests support only providers `aws`, `az`, `gcp`. It is essential to also add support for provider `local`, so that e2e tests can also test the setup and usage of local backup buckets and not just hyperscalers like AWS (through fake provider implementations such as localstack).

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
**How to categorize this PR?**

/area testing quality
/kind test

**What this PR does / why we need it**:
Add support for `Local` provider for e2e tests, along with other minor fixes/cleanups. 

**Which issue(s) this PR fixes**:
Fixes #667 

**Special notes for your reviewer**:

1. `Local` provider backup buckets are created by mounting `/etc` hostPath onto the infra job in order to create `./gardener/local-backupbuckets/<container-name>` directory and to delete it once the tests are completed. Skaffold + kustomize is used for this, similar to infra jobs of other providers like aws, azure and gcp.
2. Purging of snapshots at the beginning of each test spec was not working (`v1` snapshots were being purged instead of `v2` snapshots), due to which some tests were flaky, especially creation of 3-node etcd cluster, since backups possibly already existing in the bucket from a previous test spec. This has now been fixed, as well as enhanced for `Local` provider via a `purge-local-snapstore` job that runs before each test spec when using `Local` provider.
3. Occurrences of `etcd-main` in many tests have been replaced by `etcd-test`.
4. Repeated import of `k8s.io/api/core/v1` package as `corev1` and `v1` in `test/e2e/utils.go` has been rectified.
5. PHONY for `deploy-via-kustomize` in `Makefile` has been fixed.
6. Update doc to include `Local` provider for e2e tests; minor edits in docs too.

This PR can be tested by running:
```
make kind-up
make PROVIDERS="local" TEST_ID="e2e-test" STEPS="setup,deploy,test,undeploy,cleanup" test-e2e
make USE_ETCD_DRUID_FEATURE_GATES=true PROVIDERS="local" TEST_ID="e2e-test-2" STEPS="setup,deploy,test,undeploy,cleanup" test-e2e
make kind-down
```

**Release note**:

```feature developer
Add support for `Local` provider for e2e tests.
```

### CODE REVIEW

**Review by seshachalam-yv — APPROVED** [review]
Overall looks good to me
Thanks @shreyas-s-rao for adding local provider support to E2E test
/lgtm

File: test/e2e/utils.go [hunk 1]
```diff
@@ -628,6 +635,44 @@ func purgeSnapstore(store brtypes.SnapStore) error {
...
+							Path: "/etc",
+							Type: &directory,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:    "infra",
+					Image:   "ubuntu:23.10",
```
Comment (aaronfern) [inline test/e2e/utils.go:658]: Why use a `ubuntu` image here, won't an `alpine` image be smaller and be able to perform the same task?

**Review by aaronfern — COMMENTED** [review]
Thanks for the PR @shreyas-s-rao!
Looks good, just one question, otherwise /lgtm

File: test/e2e/utils.go (hunk 1 above)
Comment (aaronfern) [reply test/e2e/utils.go:658]: Fair enough
Keeping consistency in mind then, maybe we should (in this PR or another) have all our helper pods use `ubuntu`. In this file itself there are two more jobs being deployed (one for zero downtime validation and another for a debug pod) using different images, so maybe this is a case to have those use `ubuntu` as well

Comment (shreyas-s-rao) [reply test/e2e/utils.go:658]: https://github.com/gardener/etcd-druid/issues/670

**Review by aaronfern — APPROVED** [review]
/lgtm

**Pull request merged**